File: Algorithms/C++/00169-Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        std::unordered_map<int, int> map;

        for(int i{0}; i < nums.size(); i++){
            map[nums[i]]++;
        }
        int target = nums.size() / 2;
        for(const auto& [key, value] : map){
            if(value > target) return key;
        }

        return -1;
    }
};

/*
    tc: O(n)
    sc: O(n)

    - map frequencies
    - return the number with frequency of > n/2

*/

File: Algorithms/Python3/00169-Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counts = {}
        for num in nums:
            if num in counts:
                counts[num] = counts.get(num) + 1
            else:
                counts[num] = 1

        size = len(nums)
        for c, v in counts.items():
            if v > (size/2):
                return c

        return -1 
        
'''
    count elements in dict
    return key with value larger than size of list / 2

    tc: O(n)
    sc: O(n)
'''